src/main/java/net/fabricmc/loom/util/fmj/FabricModJsonFactory.java

```diff
@@ -166,13 +166,21 @@ public static boolean isModJar(File file, ModPlatform platform) {
 	}
 
 	public static boolean isModJar(Path input, ModPlatform platform) {
-		if (platform == ModPlatform.FORGE) {
-			return ZipUtils.contains(input, "META-INF/mods.toml");
-		} else if (platform == ModPlatform.QUILT) {
-			return ZipUtils.contains(input, "quilt.mod.json") || isModJar(input, ModPlatform.FABRIC);
-		}
+		return switch (platform) {
+			case FORGE -> ZipUtils.contains(input, "META-INF/mods.toml");
+			case QUILT -> ZipUtils.contains(input, "quilt.mod.json") || isModJar(input, ModPlatform.FABRIC);
+			default -> ZipUtils.contains(input, FABRIC_MOD_JSON);
```

Comment:
No, please don't modify this line as changes to upstream code like this are prone to cause merge conflicts. The if-else chain is fine.